src/gui/sharedialog.cpp

```diff
@@ -558,6 +572,46 @@ void ShareDialog::slotPushButtonCopyLinkPressed()
     clipboard->setText(_shareUrl);
 }
 
+void ShareDialog::slotCheckBoxEditingClicked()
+{
+    ShareDialog::setPublicUpload(_ui->checkBox_editing->checkState() == Qt::Checked);
+}
+
+void ShareDialog::setPublicUpload(bool publicUpload)
+{
+    _ui->checkBox_editing->setEnabled(false);
+    _pi_editing->startAnimation();
+
+    QUrl url;
+    QList<QPair<QString, QString> > requestParams;
+    QByteArray verb("PUT");
+
+    QString value = QString::fromLatin1(publicUpload ? "true" : "false");
+
+    url = Account::concatUrlPath(_account->url(), QString("ocs/v1.php/apps/files_sharing/api/v1/shares/%1").arg(_public_share_id));
+    requestParams.append(qMakePair(QString::fromLatin1("publicUpload"), value));
+
+    OcsShareJob *job = new OcsShareJob(verb, url, _account, this);
+    job->setPostParams(requestParams);
+    connect(job, SIGNAL(jobFinished(QVariantMap)), this, SLOT(slotPublicUploadSet(QVariantMap)));
+
+    job->start();
+}
+
+void ShareDialog::slotPublicUploadSet(const QVariantMap &reply)
+{
+    qDebug() << Q_FUNC_INFO << reply;
+
+    QString message;
+    int code = getJsonReturnCode(reply, message);
+    if (code != 100) {
+        displayError(code);
+    }
+
+    _ui->checkBox_editing->setEnabled(true);
```

Comment:
If the code was != 100 (error case) is than still good to enable the editing checkbox here?